devtools/buildVisualStories.js

```diff
@@ -0,0 +1,63 @@
+const { lstatSync, readdirSync, writeFileSync } = require('fs');
+const path = require('path');
+
+const srcPath = path.join(__dirname, '../src');
+
+const isComponentDirectory = (source) => {
+    const ignoredDirectories = ['utils', 'Docs'];
+    return lstatSync(source).isDirectory() && !ignoredDirectories.some(ignored => source.includes(ignored));
+};
+
+const componentDirs = readdirSync(srcPath).map(name => path.join(srcPath, name)).filter(isComponentDirectory).map(directory => {
+    return {
+        path: `${directory}/__stories__/`,
+        fileNames: readdirSync(`${directory}/__stories__/`)
+    };
+});
+
+// For every component directory.
+componentDirs.map((directory) => {
+    let fileContents = '';
+    let componentName = '';
+    // Loop through its files.
+    directory.fileNames.map((fileName) => {
+        // get only stories.js files
+        if (fileName.includes('.stories.js')) {
+        // Grab the component name
+            componentName = fileName.substr(0, fileName.indexOf('.'));
+            // TO DO: reenable storyshots for examples using hooks in storybook@6
+            // https://github.com/storybookjs/storybook/releases/tag/v6.0.0-alpha.43
+            if (componentName === 'Calendar'
+            || componentName === 'Dialog'
+            || componentName === 'Table'
+            || componentName === 'Popover'
+            || componentName === 'Shellbar') {
+                return;
+            }
+
+            fileContents += `
```

Comment:
Maybe we could do
```js
let fileContents = `
import React from 'react';
export default {
    title: 'Visual'
};
`
```
Then
```js
fileContents += `
import * as stories from './${componentName}.stories';
export const ${componentName} = () => {
    let storyNames = Object.keys(stories).filter(story => story !== 'default');

    return (<>{storyNames.map(item => stories[item]())}</>);
};
${componentName}.story = {
    parameters: {
        docs: { disable: true }
    }
};
`
```
Otherwise we would import React more than once and that would throw an error

Reply:
I'm confused - this works as it is? Where are you seeing it import react more than once?

Reply:
Oh I think I see where the confusion is - I'm not making one file here - I'm making one file per component. So I need that import inside the loop

Reply:
Oh, that makes sense.  Then componentName and fileContents could just be local variables in the inner loop scope
```
componentDirs.map((directory) => {
    // Loop through its files.
    directory.fileNames.map((fileName) => {
        // get only stories.js files
        if (fileName.includes('.stories.js')) {
        // Grab the component name
            const componentName = fileName.substr(0, fileName.indexOf('.'));
            // TO DO: reenable storyshots for examples using hooks in storybook@6
            // https://github.com/storybookjs/storybook/releases/tag/v6.0.0-alpha.43
            if (componentName === 'Calendar'
            || componentName === 'Dialog'
            || componentName === 'Table'
            || componentName === 'Popover'
            || componentName === 'Shellbar') {
                return;
            }

            const fileContents = `
import React from 'react';
import * as stories from './${componentName}.stories';

export default {
    title: 'Visual'
};

export const ${componentName} = () => {
    let storyNames = Object.keys(stories).filter(story => story !== 'default');

    return (<>{storyNames.map(item => <div>{stories[item]()}</div>)}</>);
};
${componentName}.story = {
    parameters: {
        docs: { disable: true }
    }
};
`;
            // write the visual story file into the directory.
            let visualPath = path.join(directory.path, `${componentName}.visual.js`);
            writeFileSync(visualPath, fileContents);
        }
    });
});
```

Reply:
Sure - do you want to push an update and test that code?

Reply:
9b8f81e

Reply:
Thank you for the help! 🙇